packages/ui-components/src/components/Chart/Table/useChartTable.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { useTable, useSortBy } from 'react-table';
+import { formatDataValueByType } from '@tupaia/utils';
+import { formatTimestampForChart, getIsTimeSeries } from '../utils';
+import { parseChartConfig } from '../parseChartConfig';
+import { CHART_TYPES, DEFAULT_DATA_KEY } from '../constants';
+
+// For the rowData, ignore labelType and use percentage instead of fractionAndPercentage as
+// we don't want to show multiple values a table cell
+const sanitizeValueType = valueType =>
+  valueType === 'fractionAndPercentage' ? 'percentage' : valueType;
+
+const getFormattedValue = (value, valueType) =>
+  value === undefined ? 'No Data' : formatDataValueByType({ value }, sanitizeValueType(valueType));
+
+const FirstColumnCell = styled.span`
+  font-weight: 500;
+  text-align: left;
+`;
+
+const makeFirstColumn = (header, accessor) => ({
+  Header: header,
+  accessor,
+  // eslint-disable-next-line react/prop-types
+  Cell: ({ value }) => <FirstColumnCell>{String(value)}</FirstColumnCell>,
+});
+
+/**
+ * Get columns to render in table
+ * Use the keys in chartConfig to determine which columns to render, and if chartConfig doesn't exist
+ * use value as the only column
+ */
+const processColumns = viewContent => {
+  const { data, xName, periodGranularity } = viewContent;
+  const hasNamedData = data[0]?.name;
+  const hasTimeSeriesData = getIsTimeSeries(data) && periodGranularity;
+  let firstColumn = null;
+
+  if (hasNamedData) {
+    firstColumn = makeFirstColumn(xName || 'Name', '');
+  }
+
+  if (hasTimeSeriesData) {
+    firstColumn = makeFirstColumn(xName || 'Date', row =>
+      formatTimestampForChart(row.timestamp, periodGranularity),
+    );
+  }
+
+  const chartConfig = parseChartConfig(viewContent);
+
+  if (Object.keys(chartConfig).length === 0) {
+    const valueColumn = {
+      Header: 'Value',
+      accessor: row => getFormattedValue(row[DEFAULT_DATA_KEY], viewContent.valueType),
+    };
+
+    return firstColumn ? [firstColumn, valueColumn] : [valueColumn];
+  }
```

Comment:
Instead of having this separate conditional block, we could use the same pattern as `CartesianChart`
```suggestion
  const chartDataConfig = Object.keys(chartConfig).length > 0 ? chartConfig : { [DEFAULT_DATA_KEY]: {} };
```
And if `parseChartConfig` is only used in a couple of places, we could then move that defaulting right inside `parseChartConfig`

Reply:
I'm not quite sure I understand. This processing seems too different to the CartesianChart to use the same code. Can you explain a bit more?

Reply:
I'll have to work out what I was meaning two weeks ago first! 😅

Reply:
Cool just re-read it and can give you a call if you like?

Reply:
Summary though, is that the processing here is actually able to be handled by the block below if we set up a default "value" column if there's no chartConfig defined. That way we can delete this whole section.

It looks like the lack of any config is meaningful in other places though so unfortunately we can't push the defaulting into `parseChartConfig` without making more widespread changes, so leave that bit of my suggestion out.

Reply:
ok I think that makes sense. I might come back to this in the morning.